src/main/java/reposense/git/GitRevList.java

```diff
@@ -57,15 +57,28 @@ public static String getCommitHashInRange(String root, String branchName, String
             return startHash;
         }
 
-        Path rootPath = Paths.get(root);
-        String revListCommand = "git rev-list " + startHash + ".." + endHash + " " + branchName
-                + REVISION_PATH_SEPARATOR;
-        String output = runCommand(rootPath, revListCommand);
+        String fromStartHash = getAllCommitHashSince(root, branchName, startHash);
+        String fromEndHash = getAllCommitHashSince(root, branchName, endHash);
+        StringBuilder output = new StringBuilder();
 
-        if (output.equals("")) {
-            return endHash + "\n" + startHash;
+        if (fromStartHash.length() > fromEndHash.length()) {
+            output.append(fromStartHash.substring(fromEndHash.length()));
+            output.append(startHash);
         } else {
-            return output + startHash + "\n";
+            output.append(fromEndHash.substring(fromStartHash.length()));
```

Comment:
same goes for this, should this be `fromEndHash.substring(0, fromEndHash - fromStartHashLength)` instead?